src/megaclient.cpp

```diff
@@ -13603,12 +13603,12 @@ bool MegaClient::startxfer(direction_t d, File* f, DBTableTransactionCommitter&
                 dstime timeleft = dstime(overquotauntil - Waiter::ds);
                 t->failed(API_EOVERQUOTA, committer, timeleft);
             }
-            else if (d == PUT && ststatus == STORAGE_RED)
+            else if (d == PUT && ststatus == STORAGE_RED && isPrivateNode(f->h))
```

Comment:
Do not fail the transfer if the upload targets a foreign account (incoming share), since we don't know whether the other account is storage overquota or not (only API knows).

Reply:
In this case we are trying to add a new file (with it's target) into an existing transfer. If target is private we need to handle as a local overquota. If the transfer also contains foreign targets, these targets must have been processed when they added and the last parameter in transfer::failed call will prevent to remove any foreign target.